src/sentry/nodestore/django/backend.py

```diff
@@ -1,49 +1,58 @@
 from __future__ import absolute_import
 
 import math
+import logging
+import pickle
 
 from django.utils import timezone
 
 from sentry.db.models import create_or_update
 from sentry.nodestore.base import NodeStorage
+from sentry.utils.strings import decompress, compress
 
 from .models import Node
 
+logger = logging.getLogger("sentry")
+
 
 class DjangoNodeStorage(NodeStorage):
     def delete(self, id):
         Node.objects.filter(id=id).delete()
         self._delete_cache_item(id)
 
-    def get(self, id):
-        item_from_cache = self._get_cache_item(id)
-        if item_from_cache:
-            return item_from_cache
+    def _decode(self, value, subkey):
+        if value is None:
+            return None
+
+        try:
+            if value.startswith(b"{"):
+                return NodeStorage._decode(self, value, subkey=subkey)
+
+            if subkey is None:
+                return pickle.loads(value)
+
+            return None
```

Comment:
I wonder how can we hit this place of code. What would be the use case?

Reply:
if one attempts to read a subkey from an event that is stored as pickle we return None here

Reply:
ah, I see. This is consistent with what we return for subkey that's not present in the new format (https://github.com/getsentry/sentry/pull/23189/files#diff-46df09659fac37c5fe7c651f49ef9f05174c596fb5166636a2afa088e30b064bR83).
Thanks.